File: packages/android-im-ui/src/test/kotlin/com/flare/im/ui/MessageLifecycleTest.kt
package com.flare.im.ui

import kotlin.test.Test
import kotlin.test.assertEquals

class MessageLifecycleTest {
    @Test
    fun lifecycleProjectsToCompleteVisualStatus() {
        assertEquals(FlareMessageDeliveryStatus.Sent, FlareMessageLifecycle().visualStatus)
        assertEquals(
            FlareMessageDeliveryStatus.Failed,
            FlareMessageLifecycle(transfer = FlareTransferState.Failed).visualStatus,
        )
        assertEquals(
            FlareMessageDeliveryStatus.Read,
            FlareMessageLifecycle(read = FlareMessageReadState.Read).visualStatus,
        )
        assertEquals(
            FlareMessageDeliveryStatus.Sending,
            FlareMessageLifecycle(send = FlareMessageSendState.Sending).visualStatus,
        )
        assertEquals(
            FlareMessageDeliveryStatus.Delivered,
            FlareMessageLifecycle(delivery = FlareMessageLifecycleDeliveryState.Delivered).visualStatus,
        )
    }
}

File: packages/flutter-im-ui/test/message_lifecycle_test.dart
import 'package:flare_im_ui/flare_im_ui.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  test('message lifecycle projects to a complete visual status', () {
    expect(
      const FlareMessageLifecycle().visualStatus,
      FlareMessageDeliveryStatus.sent,
    );
    expect(
      const FlareMessageLifecycle(
        transfer: FlareTransferState.failed,
      ).visualStatus,
      FlareMessageDeliveryStatus.failed,
    );
    expect(
      const FlareMessageLifecycle(
        read: FlareMessageReadState.read,
      ).visualStatus,
      FlareMessageDeliveryStatus.read,
    );
    expect(
      const FlareMessageLifecycle(
        send: FlareMessageSendState.sending,
      ).visualStatus,
      FlareMessageDeliveryStatus.sending,
    );
    expect(
      const FlareMessageLifecycle(
        delivery: FlareMessageLifecycleDeliveryState.delivered,
      ).visualStatus,
      FlareMessageDeliveryStatus.delivered,
    );
  });
}
